src/workerd/api/gpu/gpu-device.c++

```diff
@@ -112,18 +110,16 @@ wgpu::MipmapFilterMode parseMipmapFilterMode(kj::StringPtr mode) {
 jsg::Ref<GPUSampler> GPUDevice::createSampler(GPUSamplerDescriptor descriptor) {
   wgpu::SamplerDescriptor desc{};
 
-  desc.addressModeU = parseAddressMode(
-      descriptor.addressModeU.orDefault([] { return "clamp-to-edge"_kj; }));
-  desc.addressModeV = parseAddressMode(
-      descriptor.addressModeV.orDefault([] { return "clamp-to-edge"_kj; }));
-  desc.addressModeW = parseAddressMode(
-      descriptor.addressModeW.orDefault([] { return "clamp-to-edge"_kj; }));
-  desc.magFilter = parseFilterMode(
-      descriptor.magFilter.orDefault([] { return "nearest"_kj; }));
-  desc.minFilter = parseFilterMode(
-      descriptor.minFilter.orDefault([] { return "nearest"_kj; }));
-  desc.mipmapFilter = parseMipmapFilterMode(
-      descriptor.mipmapFilter.orDefault([] { return "nearest"_kj; }));
+  desc.addressModeU =
+      parseAddressMode(descriptor.addressModeU.orDefault([] { return "clamp-to-edge"_kj; }));
```

Comment:
If these are going to just be literal kj::StringPtrs, then using the `descriptor.addressModeU.orDefault("clamp-to-edge"_kj")` form is just fine. You want to use the callback variant only when there's an allocation cost to the default (e.g. like heap allocating a string with `kj::str(...)`)

Reply:
This is the compiler trolling me.

descriptor.addressModeU.orDefault("clamp-to-edge"_kj)
```
common.h(1460, 6): Candidate function not viable: no known conversion from 'kj::StringPtr' to 'kj::String &' for 1st argument
common.h(1467, 12): Candidate function not viable: no known conversion from 'kj::StringPtr' to 'const kj::String' for 1st argument
```
descriptor.addressModeU.orDefault([] { return kj::str("clamp-to-edge"); })
```
common.h(1460, 6): Candidate function not viable: no known conversion from '(lambda at /Users/acruz/work/vc/workerd/src/workerd/api/gpu/gpu-device.c++:114:58)' to 'kj::String &' for 1st argument
common.h(1467, 12): Candidate function not viable: no known conversion from '(lambda at /Users/acruz/work/vc/workerd/src/workerd/api/gpu/gpu-device.c++:114:58)' to 'const kj::String' for 1st argument
```

Reply:
ah right, ok.